### PR TITLE
Add LinkAttrs constructor NewLinkAttrs

### LINKED ISSUE #26: TxQLen is zero by default
But actually seems like it should be 1000. I'm not sure how I can fix this, should I implement `SetTxQLen` and move it out of `Attrs`?
@vishvananda WDYT?

### PR DESCRIPTION
**Pull request by LK4D4**
It looks little awkward, but should be enough for people who care about
defaults from kernel.

Fix #26

### CODE REVIEW

**Comment by LK4D4** [comment]
ping @vishvananda @eyakubovich

File: link.go [hunk 1]
```diff
@@ -28,6 +28,14 @@ type LinkAttrs struct {
...
 }
 
+// NewLinkAttrs returns LinkAttrs structure filled with default values
+func NewLinkAttrs() LinkAttrs {
+	return LinkAttrs{
+		TxQLen: -1,
+		MTU:    -1,
+	}
+}
+
```
Comment (eyakubovich) [inline link.go]: You're checking for `> 0` below so why not have "not set" be `== 0`. This way, default initialization still works and `NewLinkAttrs()` is not needed.

**Comment by eyakubovich** [comment]
+1 for checking for `> 0` before setting the attribute.

File: link.go (hunk 1 above)
Comment (LK4D4) [inline link.go]: Hmm, my mistake. I need `>=0`. Is it true for MTU too?

Comment (LK4D4) [inline link.go]: No, I got it. For `MTU` 0 is sorta invalid.

Comment (eyakubovich) [inline link.go]: `TxQLen` can be 0?

Comment (LK4D4) [inline link.go]: Yeah, at least I can set it with `iproute2`.

Comment (LK4D4) [inline link.go]: But that's weird, I can't set 0 actually with our lib :/

Comment (eyakubovich) [inline link.go]: Looks like TxQLen == 0 has important special meaning in some drivers. Actually, same for MTU (basically says default to lower dev MTU). I guess we need `NewLinkAttrs`. @vishvananda thoughts?

Comment (LK4D4) [inline link.go]: Hmm, I'm not sure, but seems like we can't set `TxQLen` to `0` with netlink. `iproute2` uses `ioctl`.

Comment (vishvananda) [inline link.go]: If we can't actually set zero for either then we could just leave zero as
the default, but I thought the initial bug was that it doesn't default to
10000

Vish

Comment (LK4D4) [inline link.go]: https://github.com/torvalds/linux/blob/master/net/core/rtnetlink.c#L1702

Comment (LK4D4) [inline link.go]: I was pretty sure, that we can :/ Actually with `TxQLen` all works as expected, apart from setting it to `0`.

Comment (eyakubovich) [inline link.go]: But from that kernel code, it looks like 0 is ok and should be set. `tb[IFLA_TXQLEN` is just `rtattr *`

Comment (LK4D4) [inline link.go]: Yes, sorry, I'll retry my tests. TxQLen should be set.

**Comment by LK4D4** [comment]
Okay, I think I found it. For veth pair, `TxQLen` is set to 1000 by default, but with netlink you're changing only peer `TxQLen`. For `dummy` type default is 0.

**Comment by LK4D4** [comment]
@eyakubovich @vishvananda Thanks for help!
Seems like it never worked! I changed code and added some tests.
MTU seems can't be zero, it returns `EINVAL` for Veth creation for example.

**Comment by eyakubovich** [comment]
With veth, the main device and its peer are two separate devices (they each have ifindex). So you can't change TxQLen on them individually? Or did you mean that at creation time, TxQLen affects only peer.

**Comment by LK4D4** [comment]
No, I was wrong. Logic of txqlen was broken for main device. 
Also you can set different qlens for main and peer,  you can try with iproute2.

**Comment by LK4D4** [comment]
And original test for txqlen was false positive. I think all should be OK now.

**Comment by vishvananda** [comment]
The code in the tests and the readme should be changed to use NewLinkAttrs. Unfortunately this means the nice one liners have to go away.

One option for allowing one liners would be to allow the New to take and return a pointer to a LinkAttrs that modifies the default:

http://play.golang.org/p/d4ZYrukfDE

but it has the drawback that NewLinkAttrs(&LinkAttrs{TxQLen: 0}) doesn't work as expected.

**Comment by LK4D4** [comment]
People still can use oneliners if they will set TxQLen explicitly. I'll write about this in Readme.

**Comment by LK4D4** [comment]
@vishvananda I added info to readme, but changing tests is a lot of work and not really worth, because we really don't care about `TxQLen` in most tests.

**Pull request merged**